### PR TITLE
Interp test reorg

### PR DESCRIPTION
**Pull request by rachitnigam**
While cleaning up the test suite for #575, I took the opportunity to clean up
and reorganize the test suite.

The `tests/control/iteration` tests are disabled for the lowered tests because
of the bug mentioned in #575.

### CODE REVIEW

**Review by sampsyo — COMMENTED** [review]
Love it. It would be great to get confirmation from at least some of @almathaler, @ai-cu, and @yn224—does this reorganization make sense to y'all too? (You can look at just the `runt.toml` in this PR to get a sense of what's changed.)

File: interp/runt.toml [hunk 1]
```diff
@@ -1,41 +1,42 @@
...
-name = "Simple tests"
+name = "unit"
 paths = [
-    "tests/simple_tests/*.futil"
+  "tests/unit/*.futil"
 ]
-
 cmd = """
-../target/debug/interp {}
+  ../target/debug/interp {} | jq .memories
```
[the remark this answers is not in the document]
Comment (EclecticGriffin) [reply interp/runt.toml:10]: Yeah this makes sense given that most of the ports no longer show anything interesting in the final state. We may wish to add a means of printing out intermediate state later

File: interp/runt.toml [hunk 2]
```diff
@@ -1,41 +1,42 @@
...
 paths = [
-    "tests/lowered_tests/lowered_simple_tests/*.futil"
+    "tests/control/*.futil",
+  # Disabled iteration tests due to bug
+  #    "tests/control/iteration/*.futil"
 ]
-
 cmd = """
-../target/debug/interp {}
+../target/debug/futil {} -d static-timing -l ../ | ../target/debug/interp | jq .memories
```
Comment (EclecticGriffin) [inline interp/runt.toml:40]: Love it!

File: interp/runt.toml [hunk 3]
```diff
@@ -1,41 +1,42 @@
 ver = "0.2.8"
 
+# Check basic functionality of the interpreter
 [[tests]]
-name = "Simple tests"
+name = "unit"
 paths = [
-    "tests/simple_tests/*.futil"
+  "tests/unit/*.futil"
```
Comment (EclecticGriffin) [inline interp/runt.toml:7]: 👍🏻

**Comment by rachitnigam** [comment]
Can someone try to resolve the conflicts and merge (maybe @EclecticGriffin). If you can’t, I’ll get to it tomorrow.

**Comment by EclecticGriffin** [comment]
Sure, happy to take the metaphorical lock on this

**Comment by EclecticGriffin** [comment]
Conflicts dealt with

**Comment by EclecticGriffin** [comment]
Dealt with again, worth noting that one of the tests `register-writes-visible-inside-group.futil` doesn't work (infinite loop) because the program has a bug which didn't become a problem until the zero-assigns PR was merged.

Also apologies for the git mess, not sure why it was giving me such a hard time

**Comment by rachitnigam** [comment]
What’s the bug and can we fix the test so that we can merge?

**Comment by rachitnigam** [comment]
I've fixed the test, merged in old tests, and saved the output of error tests.

**Pull request merged**